subgraphs/compound-v3/src/sdk/manager.ts

```diff
@@ -0,0 +1,902 @@
+    withdraw.gasLimit = this.event.transaction.gasLimit;
+    withdraw.blockNumber = this.event.block.number;
+    withdraw.timestamp = this.event.block.timestamp;
+    withdraw.account = account;
+    withdraw.market = this.market.id;
+    withdraw.position = positionID!;
+    withdraw.asset = asset;
+    withdraw.amount = amount;
+    withdraw.amountUSD = amountUSD;
+    withdraw.save();
+
+    this.snapshots.updateTransactionData(
+      TransactionType.WITHDRAW,
+      amount,
+      amountUSD
+    );
+    this.snapshots.updateUsageData(TransactionType.WITHDRAW, account);
+
+    return withdraw;
+  }
+
+  createBorrow(
+    asset: Bytes,
+    account: Bytes,
+    amount: BigInt,
+    amountUSD: BigDecimal,
+    newBalance: BigInt,
+    tokenPriceUSD: BigDecimal | null = null, // used for different borrow token
+    interestType: string | null = null
+  ): Borrow {
+    const borrower = new AccountManager(
+      account,
+      this.market,
+      this.protocol,
+      this.event
+    );
+    if (borrower.isNewAccount()) {
+      this.protocol.cumulativeUniqueUsers += INT_ONE;
+      this.protocol.save();
+    }
+    borrower.addPosition(
+      asset,
+      newBalance,
+      PositionSide.BORROWER,
+      TransactionType.BORROW,
+      tokenPriceUSD ? tokenPriceUSD : this.market.inputTokenPriceUSD,
+      interestType
+    );
+
+    const borrow = new Borrow(
+      this.event.transaction.hash
+        .concatI32(this.event.logIndex.toI32())
+        .concatI32(Transaction.BORROW)
+    );
+    borrow.hash = this.event.transaction.hash;
+    borrow.nonce = this.event.transaction.nonce;
+    borrow.logIndex = this.event.logIndex.toI32();
+    borrow.gasPrice = this.event.transaction.gasPrice;
+    borrow.gasUsed = this.event.receipt ? this.event.receipt!.gasUsed : null;
+    borrow.gasLimit = this.event.transaction.gasLimit;
+    borrow.blockNumber = this.event.block.number;
+    borrow.timestamp = this.event.block.timestamp;
+    borrow.account = account;
+    borrow.market = this.market.id;
+    borrow.position = borrower.getPositionID()!;
+    borrow.asset = asset;
+    borrow.amount = amount;
+    borrow.amountUSD = amountUSD;
+    borrow.save();
+
+    this.snapshots.updateTransactionData(
+      TransactionType.BORROW,
+      amount,
+      amountUSD
+    );
+    this.snapshots.updateUsageData(TransactionType.BORROW, account);
+
+    return borrow;
+  }
+
+  createRepay(
+    asset: Bytes,
+    account: Bytes,
+    amount: BigInt,
+    amountUSD: BigDecimal,
+    newBalance: BigInt,
+    tokenPriceUSD: BigDecimal | null = null, // used for different borrow token
+    interestType: string | null = null
+  ): Repay | null {
+    const repayer = new AccountManager(
+      account,
+      this.market,
+      this.protocol,
+      this.event
+    );
+    if (repayer.isNewAccount()) {
+      this.protocol.cumulativeUniqueUsers += INT_ONE;
+      this.protocol.save();
+    }
```

Comment:
As an alternative, you could potentially embed these bits into the account manager. It would not work though if we counted independent type of users (lps, borrowers, lenders, etc)

Reply:
I did that bc sometimes you make an account but don't count it, like the receiver in a transaction. Bc we are only counting users that spend gas.

But that opens the door to missing users if they interact w the protocol after the fact...

Reply:
Interesting ... Sounds like we should fix that then :P 
Maybe by only counting the users when increasing account counters? When increasing `Account.depositCount` (and the rest of counters), do that from the AccountManager, and if all counters are at `0` then you can increase the counter of unique total users.

Reply:
Fixed